pkg/routeagent/controllers/route/route.go

```diff
@@ -526,9 +534,8 @@ func (r *Controller) handleRemovedEndpoint(obj interface{}) {
 			klog.Errorf("Could not convert object tombstone %v to an Endpoint", tombstone.Obj)
 			return
 		}
-		klog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
-	klog.V(4).Infof("Informed of removed endpoint for route controller object: %v", object)
+	klog.V(log.DEBUG_LEVEL).Infof("Informed of removed endpoint for route controller object: %v", object)
 	hostname, err := os.Hostname()
```

Comment:
Thanks to @tpantelis for pointing this out. 

If we are on the gateway node, it's a good idea to cleanup the xfrm policies as well.
i.e., r.cleanXfrmPolicies

Reply:
Small note: Actually the r.cleanRoutes() is not required in this API because as soon as we delete the VxLAN interface the routes associated on that interface will be deleted. I'm fine if you want to remove that call from this API.

Reply:
no, if it's active GW we should not clean xfrm policies (those are set by strongswan). The other routes yes, because those are the routes which would be used on the non-gw node.
I suggest, that after release we could perform a small refactor of this agent to make it clearer. Probably smaller functions and isolating better the GW / non-gw behaviors/details.

Reply:
yeah I've already have patches lined up and other changes planned.

Reply:
> Small note: Actually the r.cleanRoutes() is not required in this API because as soon as we delete the VxLAN interface the routes associated on that interface will be deleted. I'm fine if you want to remove that call from this API.

Let's leave it there for now, I think it will be better not to delete those interfaces where we don absolutely need- I want to propose a patch after the release.